server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -101,7 +101,8 @@ public SegmentLoaderLocalCacheManager(
           )
       );
     }
-    locations.sort(COMPARATOR);
+    this.strategy = config.getStorageLocationSelectorStrategy();
+    this.strategy.setStorageLocations(ImmutableList.copyOf(locations));
```

Comment:
`locations` are effectively immutable. Don't need to make a copy of it.

Reply:
This line is removed as `setStorgaeLocations()` method is removed from `StorageLocationSelectorStrategy` interface.